api-reference/10 UI Widgets/GridBase/1 Configuration/scrolling/rowRenderingMode.md

```diff
@@ -17,6 +17,9 @@ Renders all the loaded rows at once.
 - *"virtual"*     
 Renders only those rows that get into the viewport. Use this mode if the [pageSize](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/pageSize.md '{basewidgetpath}/Configuration/paging/#pageSize') is large. Rows that are being currently rendered can be shown as grey boxes.
 
+    [note] While scrolling in **virtual** mode, gray boxes can appear when the widget renders data, in order to avoid freezes and provide smooth scrolling. It happens when the data is being loaded from a remote data source or when a local data source has a heavy structure, for example, it has templates, too many columns for your device, etc.     
```

Comment:
1. Grey/gray boxes are already mentioned in the previous sentence:

    > Rows that are being currently rendered can be shown as grey boxes.

    You should reorder the information.

2. So, the gray boxes appear in virtual _scrolling_ mode. But they are mentioned in the description of the virtual _rendering_ mode. Clarify.

Reply:
This information is suitable for both rendering and scrolling virtual modes. There are too many questions from clients about these gray boxes that's why it's necessary to add a note explaining this.

I rewrote the article, is it okay now?